provider/datagen/src/uprops/casemapping.rs

```diff
@@ -3,24 +3,32 @@
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
 use crate::uprops::uprops_serde;
+use crate::DatagenOptions;
 use icu_casemapping::provider::{CaseMappingV1, CaseMappingV1Marker};
 use icu_casemapping::CaseMappingInternals;
 use icu_codepointtrie::CodePointTrieHeader;
 use icu_provider::prelude::*;
 
 use std::convert::TryFrom;
 use std::fs;
-use std::path::PathBuf;
 
 pub struct CaseMappingDataProvider {
     case_mapping: CaseMappingInternals<'static>,
 }
 
 /// A data provider reading from .toml files produced by the ICU4C icuwriteuprops tool.
-impl CaseMappingDataProvider {
-    pub fn try_new(path: PathBuf) -> Result<Self, DataError> {
+impl TryFrom<&DatagenOptions> for CaseMappingDataProvider {
+    type Error = DataError;
+    fn try_from(options: &DatagenOptions) -> Result<Self, Self::Error> {
```

Comment:
Nit (optional): use the same style as the other `TryFrom<&DatagenOptions>` where there is a `try_new` that takes non-optional arguments and `TryFrom` delegates to it

Reply:
Also in my follow-up